### PR TITLE
update code to reflect latest changes as of May 30th

### PR DESCRIPTION
**Pull request by prathikr**
updates code to reflect latest changes as of May 30th

### CODE REVIEW

**Comment by prathikr** [comment]
@patrickvonplaten @anton-l can I please get a review on this PR updating the 3 tasks under examples/research_projects/onnxruntime? Thank you!

File: examples/research_projects/onnxruntime/textual_inversion/README.md [hunk 1]
```diff
@@ -53,20 +53,29 @@ If you have already cloned the repo, then you won't need to go through these ste
...
 
 ```bash
-export MODEL_NAME="runwayml/stable-diffusion-v1-5"
-export DATA_DIR="path-to-dir-containing-images"
-
 accelerate launch textual_inversion.py \
-  --pretrained_model_name_or_path=$MODEL_NAME \
-  --train_data_dir=$DATA_DIR \
+  --pretrained_model_name_or_path="runwayml/stable-diffusion-v1-5" \
+  --train_data_dir="path-to-dir-containing-images" \
```
Comment (bghira) [inline examples/research_projects/onnxruntime/textual_inversion/README.md:78]: why?

Comment (prathikr) [reply examples/research_projects/onnxruntime/textual_inversion/README.md]: removed, thank you for catching that

**Review by bghira — APPROVED** [review]
thanks for bringing that variable back

File: examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py [hunk 2]
```diff
@@ -28,30 +29,96 @@
...
 from diffusers import AutoencoderKL, DDPMScheduler, StableDiffusionPipeline, UNet2DConditionModel
 from diffusers.optimization import get_scheduler
 from diffusers.training_utils import EMAModel
-from diffusers.utils import check_min_version
+from diffusers.utils import check_min_version, deprecate, is_wandb_available
 from diffusers.utils.import_utils import is_xformers_available
 
 
+if is_wandb_available():
+    import wandb
```
Comment (bghira) [inline examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:53]: might want to raise an error if there's a request to use wandb tracker and it's not available

Comment (prathikr) [reply examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:53]: @bghira I'm trying to keep this example consistent with the non-optimized example under `diffusers/examples/text_to_image/train_text_to_image.py` 

Should this change be made there as well? Is it best to add that to this PR or open a new one adjusting both scripts?

Comment (bghira) [reply examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:53]: i'm not used to reviewing stuff or how code style works here, so please take my suggestions with a grain of salt, that's just how we do things in my day-to-day. if it's something we want to do, maybe i can take a stab at it later in follow-up PRs.

Comment (prathikr) [reply examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:53]: sounds good, will resolve this thread and leave it to you to update the scripts later if necessary

File: examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py [hunk 3]
```diff
@@ -383,17 +509,81 @@ def main():
...
+                model.register_to_config(**load_model.config)
+
+                model.load_state_dict(load_model.state_dict())
+                del load_model
+
+        accelerator.register_save_state_pre_hook(save_model_hook)
+        accelerator.register_load_state_pre_hook(load_model_hook)
+
     if args.gradient_checkpointing:
         unet.enable_gradient_checkpointing()
```
Comment (bghira) [inline examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:586]: are we not training the text encoder when fine-tuning? i do, but i freeze most of the layers. am i not supposed to?

File: examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py [hunk 4]
```diff
@@ -660,8 +862,24 @@ def collate_fn(examples):
                     raise ValueError(f"Unknown prediction type {noise_scheduler.config.prediction_type}")
 
                 # Predict the noise residual and compute loss
-                model_pred = unet(noisy_latents, timesteps, encoder_hidden_states, return_dict=False)[0]
-                loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
+                model_pred = unet(noisy_latents, timesteps, encoder_hidden_states).sample
+
+                if args.snr_gamma is None:
+                    loss = F.mse_loss(model_pred.float(), target.float(), reduction="mean")
```
Comment (bghira) [inline examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:868]: in dreambooth script, we find the target before the prediction. here, we do the opposite. does it matter?

File: examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py (hunk 3 above)
Comment (prathikr) [reply examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:586]: @bghira again, I would like to keep this consistent with the existing examples.

File: examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py (hunk 4 above)
Comment (prathikr) [reply examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py:868]: same comment as before

**Review by anton-l — APPROVED** [review]
Thanks a lot for maintaining the examples @prathikr!
